### PR TITLE
[type:refactor]Decouple apidoc module from rule and selector modules

### PR DESCRIPTION
**Pull request by 478320**
decouple apidoc module from rule and selector modules, link https://github.com/apache/shenyu/issues/5623

Make sure that:
 - [X] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
 - [X] You submit test cases (unit or integration tests) that back your changes.
 - [X] Your local test passed ./mvnw clean install -Dmaven.javadoc.skip=true.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java [hunk 1]
```diff
@@ -110,6 +109,7 @@ public String createOrUpdate(final ApiDTO apiDTO) {
      * @return update message
      */
     private String update(final ApiDTO apiDTO) {
+        //cdbuild apiDO
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:112]: why add this？

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java [hunk 2]
```diff
@@ -126,11 +126,6 @@ private String update(final ApiDTO apiDTO) {
                 tagRelationMapper.deleteByApiId(apiDO.getId());
                 tagRelationMapper.batchInsert(tags);
             }
-            if (ApiStateEnum.PUBLISHED.getState() == apiDO.getState()) {
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:129]: why remove this？

Comment (478320) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:129]: Hello, my previous submission failed to check the instability of ci. I hope I can re-trigger ci through some modifications. If these modifications cause bad effects, I will delete them

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java (hunk 1 above)
Comment (478320) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:112]: Hello, my previous submission failed to check the instability of ci. I hope I can re-trigger ci through some modifications. If these modifications cause bad effects, I will delete them

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java (hunk 2 above)
Comment (478320) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:129]: Hello, in order to realize that the user can directly use the api document after adding it in the visual operation interface, shenyu Gateway creates the default selector and rule when calling to create the api document, so the user can directly use the document for testing. But this also causes the api document to be coupled with selector and rule. To delete this part, users need to manually create the api document and test the need to manually add selector, rule, metadate, but solve the coupling problem of api and selector and rule. Users who use annotations such as @ShenyuGetMapping, as well as those who use swagger, will not be affected by the automatic creation of selectors and so on

Comment (478320) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:129]: When creating apidoc, this part determines whether the document is in the published state, and if the document is in the published state, the selector is registered, etc. If it is in the offline state, the corresponding selector content is deleted, which will lead to the coupling of apidoc and selector. Deleting this part can release their coupling. The removal of this feature will have the following effects, the document publication and logout will only be displayed as a status without additional functions, the document logout and online need to manually add and delete selectors and other content, for the creation of api documents, Users using @RequestMapping and other annotations and users using swagger will automatically create selectors and other content through annotations, which will not affect the convenience brought by automation. To add api documents in the visual interface, you need to manually create corresponding selector, rule, and Metadata. For the offline api document, all users need to manually delete the corresponding selector, rule, and Metadata

**Review by moremind — APPROVED** [review]

**Pull request merged**